helm-charts/common/data-prep/milvus-values.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Default values for data-prep.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+milvus:
+  enabled: true
+  cluster:
+    enabled: false
+  etcd:
+    replicaCount: 1
+  pulsar:
+    enabled: false
+  minio:
+    mode: standalone
+redis-vector-db:
```

Comment:
In the Chart.yaml, we set dependencies for all possible components. We have to specify enabled:false explicitly for each component, otherwise helm chart will install all the dependencies by default.
dependencies:
  - name: tei
    version: 1.0.0
    repository: file://../tei
    condition: tei.enabled
  - name: redis-vector-db
    version: 1.0.0
    repository: file://../redis-vector-db
    condition: redis-vector-db.enabled
  - name: milvus
    version: 4.2.12
    repository: https://zilliztech.github.io/milvus-helm/
    condition: milvus.enabled

Reply:
With the dependency and enabled falg, user would have the flexibility of 1) Using external deployed Embedding/VectorDB, or 2) Choose one set of Embedding/VectorDB and auto deployed by helm chart.